src/logic/calculate.js

```diff
@@ -54,45 +64,62 @@ export default function calculate(obj, buttonName) {
     }
     return { total: '0.' };
   }
-  if (buttonName === '+/-') {
-    const newobj = {
-      total: '',
-      next: '',
-      operation: buttonName,
-    };
 
-    if (isNumber(next) && isNumber(total)) {
+  if (buttonName === '=') {
+    if (obj.next && obj.operation) {
       return {
-        total: (-1 * parseFloat(total).toString()),
-        next: (-1 * parseFloat(next).toString()),
-        operation,
+        total: operate(obj.total, obj.next, obj.operation),
+        next: null,
+        operation: '',
       };
     }
-    if (isNumber(next)) {
-      newobj.next = (-1 * parseFloat(next).toString());
-    }
 
-    if (isNumber(total)) {
-      newobj.total = (-1 * parseFloat(total).toString());
-    }
+    // '=' with no operation, nothing to do
+    return {};
+  }
 
-    return newobj;
+  if (buttonName === '+/-') {
+    if (obj.next) {
+      return { next: (-1 * parseFloat(obj.next)).toString() };
+    }
+    if (obj.total) {
+      return { total: (-1 * parseFloat(obj.total)).toString() };
+    }
+    return {};
   }
-  if (buttonName === '=') {
-    if (isNumber(total) && !isNumber(next)) {
+
+  // Button must be an operation
+
+  // When the user presses an operation button without having entered
+  // a number first, do nothing.
+  // if (!obj.next && !obj.total) {
+  //   return {};
+  // }
```

Comment:
- [ ]  [OPTIONAL] Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.